### DISCUSSION TITLE
Failed to load and execute Flow's environment: ModuleNotFoundError ("No model named 'C'")

### DISCUSSION THREAD

**Discussion by porbmv83** [PrefectHQ/prefect#5143] [Q&A] [answered]
Hi, 
I've installed prefect in my AKS cluster. Everything seems to be up and running. When I try to run the sample flow it fails with the following message: Failed to load and execute Flow's environment: ModuleNotFoundError ("No model named 'C'") .
The sample fow:

import prefect
from prefect import task, Flow

@task
def say_hello():
    logger = prefect.context.get("logger")
    logger.info("Hello, Cloud!")

with Flow("hello-flow") as flow:
    say_hello()

flow.register(project_name="tutorial")

Could anyone provide me some hints?
Thanks

**Comment by anna-geller** [comment, marked as answer]
@porbmv83
- when you use the default Local storage when you register, and then you try to run the flow on a different machine, this is when this error usually happens
- When you register your flow and don't attach any storage explicitly, Prefect will save your flow in a Local storage file. When you run the flow, Prefect loads it from that storage file and executes it. 
- For the default Local storage, this is saved in the `~/.prefect` folder when you register.
- So when Prefect executes the content of the flow, it imports it from that Local storage. 
- The No module named 'C' is because you’re using Windows, and that is the default file drive, and the import on the Prefect end will look like “from C..prefect.flows import your-flow” 
- This specific error is just saying that it can’t find the file to load in. This happens when you register with Local storage on one machine, and then run it on another machine (e.g. on AKS or on another server) on which this flow file does not exist.

The easiest way to solve it is to use one of Git or cloud object storage classes e.g. GitHub, Bitbucket, S3, Azure storage etc. Here you can find some examples that combine with AKS: https://github.com/anna-geller/packaging-prefect-flows/tree/master/flows 
e.g. Azure storage + Azure AKS KubernetesRun run configuration https://github.com/anna-geller/packaging-prefect-flows/blob/master/flows/azure_kubernetes_run_custom_azure_image.py

**Reply by porbmv83** [reply]
@anna-geller - Thanks a lot for your reply! It works well when I use the GitHub storage.
I'm trying to test the integration with Dask using the same example but it doesn't seem to work.
I don't see any Dask works pods being initialized and run when I trigger the below flow. Would the below flow work?

from prefect import task, Flow, Parameter
import datetime
import random
from prefect.storage import GitHub
from prefect.run_configs import KubernetesRun
from prefect.environments import DaskKubernetesEnvironment

FLOW_NAME = "dask_etl"
STORAGE = GitHub(
    repo="porbmv83/prefect_test",
    path=f"{FLOW_NAME}.py",
)

@task
def inc(x):
    return x + 1


@task
def dec(x):
    return x - 1

@task
def add(x, y):
    return x + y

@task
def list_sum(arr):
    return sum(arr)

with Flow(FLOW_NAME, storage=STORAGE, run_config=KubernetesRun(labels=["porbmv"],), environment = DaskKubernetesEnvironment(min_workers=1, max_workers=3)) as flow:
    incs = inc.map(x=range(100))
    decs = dec.map(x=range(100))
    adds = add.map(x=incs, y=decs)
    total = list_sum(adds)

**Reply by anna-geller** [reply]
@porbmv83 the DaskKubernetesEnvironment has been deprecated, you should attach a Dask executor to your flow instead. You can find more about it, incl. examples here: https://docs.prefect.io/orchestration/flow_config/executors.html#daskexecutor

**Comment by ivarec** [comment]
Just a suggestion: you could catch this specific exception and give the user a message error that will make her understand immediately what's wrong :)
